standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -368,33 +368,37 @@ public TxnHandler() {
   public void setConf(Configuration conf){
     this.conf = conf;
 
+    int maxPoolSize = MetastoreConf.getIntVar(conf, ConfVars.CONNECTION_POOLING_MAX_CONNECTIONS);
+    long getConnectionTimeoutMs = 30000;
     synchronized (TxnHandler.class) {
       if (connPool == null) {
-        Connection dbConn = null;
-        // Set up the JDBC connection pool
-        try {
-          int maxPoolSize = MetastoreConf.getIntVar(conf, ConfVars.CONNECTION_POOLING_MAX_CONNECTIONS);
-          long getConnectionTimeoutMs = 30000;
-          connPool = setupJdbcConnectionPool(conf, maxPoolSize, getConnectionTimeoutMs);
-          /*the mutex pools should ideally be somewhat larger since some operations require 1
+        connPool = setupJdbcConnectionPool(conf, maxPoolSize, getConnectionTimeoutMs);
```

Comment:
I wonder if we should initialize these static fields separately (perhaps in HiveMetaStore) instead of within each handler thread. Otherwise, all the handler threads will try this code and fail inevitably.

Reply:
Do you mean eager initialization on HMS startup?  I had thought it too and initially did not want to change the behavior and add more connections for potentially nothing for HMS user does not use any Txn functions.  Is it what you mean?  (Not sure all the threads will inevitably fail though, after the first succeeds they will skip)

Reply:
> potentially add more connections for for HMS user who do not use any Txn functions

Yea that's true .. I wonder if there is a way to know that the txn feature will be used beforehand.

> Not sure all the threads will inevitably fail though, after the first succeeds they will skip

What I mean is if there is some fatal issue in the DB connection for instance, all the threads will try the same path and fail. It's better to just fail once instead.

Reply:
Yea it is maybe better..  Though I don't see anything about turning off txn handler, the only thing is :  metastore.txn.store.impl but it always has a default value.

@pvary  @deniskuzZ any thoughts if we prefer eager initialization?

Reply:
> What I mean is if there is some fatal issue in the DB connection for instance, all the threads will try the same path and fail. It's better to just fail once instead.

By the way in our case it would have recovered as db connection becomes available after awhile :) , but yea hard to generalize the case